x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -1070,10 +1162,40 @@ private void doUpdateApiKey(
         final Set<RoleDescriptor> userRoleDescriptors,
         final VersionedApiKeyDoc currentVersionedDoc,
         final ActionListener<UpdateApiKeyResponse> listener
+    ) throws IOException {
+        final IndexRequest indexRequest = maybeBuildIndexRequest(currentVersionedDoc, authentication, request, userRoleDescriptors);
```

Comment:
My plan is to follow this PR up with a refactor to port the single update method over to use `bulkUpdateApiKey` to remove the redundant code below.